### DISCUSSION TITLE
No Ruins spawn chance parameter in biomes xml / too many ruins?

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#9715] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Unlike wrecks, there is no setting to control the amount/frequency of ruin spawns.
There are no ruins for biomes 1 and 2 (which is intentional), but every single cave in biomes 3 4 5 generates ruins. 

![image](https://user-images.githubusercontent.com/73229309/181616655-f91da24c-fc8a-4bac-bcfa-05cefb309508.png)

Wrecks have a minimum/maximum while ruins are either 0 or 1.

Ruins increase loading times of biomes 3 4 5
and
Ruins always follow the same generation parameter ("Ruin" count="8"  "RuinVault" count="1""RuinEntrance" count="1"), so unlike wrecks which can be small, big or thalamused, Ruins don't have that kind of potential loot/danger variation

Use a % chance of spawning ruins setting to cut down on the amount of ruins?

**Comment by NilanthAnimosus** [comment]
I don't feel it is guaranteed that players will locate the ruins or that they will generate close to the path a submarine might actually take? reducing the amount that spawns seems odd to me (As a player).

though if it is not possible to, perhaps it should be an option for ruins to have a % chance of generating in the XMLs at the very least or a range.

As for ruin difficulty, that sounds more like a balance issue than if their too easy for risk vs reward just reduce how many exist.

**Comment by ubertpendragon** [comment]
I'm more concerned about every level from biome 3+ generating these huge structures which have to run the outpost generation logic and load several different .sub files, although I don't know which impacts loading times more:
A single somewhat large wreck file or multiple small rooms stitched together?

> I don't feel it is guaranteed that players will locate the ruins or that they will generate close to the path a submarine might actually take? 

Ruin are big and as a result map generation can lead to these kind of situations:
![image](https://user-images.githubusercontent.com/73229309/181739492-705e4dfc-0c76-42f8-8108-0687cf28891e.png)
They also have a peculiar sonar signature which makes them easy to find if not inside a cave.

Overall ruins are pretty hit or miss (for balance reasons):
Players will ignore ruins inside caves because they can't shoot through them with sub guns.
On the other hand, "open water" ruins are trivialized

I think the game doesn't need to generate ruins in every cave. The overwhelming majority of ruins will be skipped by players, unless spawned as a mission objective.
